openedx/core/djangoapps/edx_global_analytics/tests/test_acceptor_api_usage.py

```diff
@@ -0,0 +1,173 @@
+"""
+Tests acceptor API usage by edx global analytics application functions aka utils.
+"""
+
+import uuid
+import unittest
+
+from mock import patch
+
+from openedx.core.djangoapps.edx_global_analytics.utils import send_instance_statistics_to_acceptor
+from openedx.core.djangoapps.edx_global_analytics.token_utils import (
+    access_token_authorization,
+    access_token_registration,
+    get_acceptor_api_access_token,
+)
+
+
+@patch('openedx.core.djangoapps.edx_global_analytics.utils.requests.post')
+class TestAcceptorApiUsage(unittest.TestCase):
+    """
+    Test acceptor API usage by edx global analytics application functions aka utils.
+    """
+
+    def test_token_registration(self, mock_request):
+        """
+        Verify that access_token_registration sends request to acceptor API for token registration.
+        """
+        access_token_registration('https://mock-url.com')
+
+        mock_request.assert_called_once_with('https://mock-url.com' + '/api/token/registration/')
+
+    @patch('openedx.core.djangoapps.edx_global_analytics.models.AccessTokensStorage.objects.create')
+    def test_saving_token(self, mock_access_tokens_storage_model_objects_create_method, mock_request):
+        """
+        Verify that access_token_registration gets access token after registration request to acceptor and
+        save it to AccessTokensStorage model in database.
+        """
+        mock_access_token = uuid.uuid4().hex
+
+        mock_request.return_value.json.return_value = {
+            'access_token': mock_access_token
+        }
+
+        access_token_registration('https://mock-url.com')
+
+        mock_access_tokens_storage_model_objects_create_method.assert_called_with(access_token=mock_access_token)
```

Comment:
you testing here, that access token is passing to model. This is not straightforward. Maybe unit tests of functions will be enough. Can discuss it.